app/controllers/events_controller.rb

```diff
@@ -20,6 +20,17 @@ def create
 
   def show
     @event = Event.find(params[:id])
+    @status = @event.user_event_status(current_user)
+  end
+
+  def status
+    Event.change_user_event_role(current_user, params[:id][:status], params[:id][:event_id])
+    redirect_to request.referrer
+  end
+
+  def destroy
+    event = Event.find(params[:id]).destroy
```

Comment:
This is a good example of a place to use Current User rather than the event model. So like current_user.hosting.where(event_id: params[:id]).destroy